addons/mail/static/src/messaging/entity/follower/follower.js

```diff
@@ -20,11 +20,16 @@ function FollowerFactory({ Entity }) {
         static convertData(data) {
             const data2 = {};
             if ('channel_id' in data) {
-                const channelData = { id: data.channel_id, model: 'mail.channel' };
-                if ('name' in data) {
-                    channelData.name = data.name;
+                if (!data.channel_id) {
+                    data2.channel = [['unlink-all']];
+                } else {
+                    const channelData = this.env.entities.Thread.convertData({
```

Comment:
IMO this should not be converted

Reply:
And what if `channel_type` was given for example instead of `model`? Wouldn't we want model to be set automatically as per `convertData`?

Reply:
`convertData` exists just to translate from server formatter to entity fields. Here the fields to update are explicit, so it shouldn't pass into `convertData` at all.

If we really want `channel_type` and auto-update `model` accordingly in the future, then `channel_type` should be an attribute and `model` is a computed field with `channel_type` as one of its dependencies.

Reply:
As discussed, ok to remove it, `convertData` should be renamed (another task) to better convey it is actually flux specific.